### PR TITLE
Bump NFD Go to 1.18

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
https://github.com/golangci/golangci-lint/releases/tag/v1.45.0 adds support for go1.18, we are good to go
/hold cancel

File: config/jobs/image-pushing/k8s-staging-nfd-operator.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ postsubmits:
       # job makes sense on every branch (unless it's setting a `latest` tag it
       # probably does).
       branches:
+        - ^main$
```
Comment (marquiz) [inline config/jobs/image-pushing/k8s-staging-nfd-operator.yaml:18]: nit 🙄: unrelated change

Comment (ArangoGutierrez) [reply config/jobs/image-pushing/k8s-staging-nfd-operator.yaml:18]: plop why this change is here, I had a diff branch 🤔  jjejeje sorry fixing....

Comment (marquiz) [reply config/jobs/image-pushing/k8s-staging-nfd-operator.yaml]: np 😊

**Review by marquiz — APPROVED** [review]
Thanks!
/lgtm
